lib/ip.go

```diff
@@ -36,13 +37,56 @@ func RandIP4List(n int) []net.IP {
 	return ips
 }
 
+// RandIP4Range returns a list of randomly generated IPv4 addresses within
+// starting and ending IPrange.
+
+// note: starting and ending IPs should be in format of "1.1.1.1"
+func RandIP4Range(startIP, EndIP string) (net.IP, error) {
+	StartIPRaw := net.ParseIP(startIP).To4()
+	if len(StartIPRaw) == 0 || len(StartIPRaw) > net.IPv4len {
+		return nil, fmt.Errorf("range is Invalid")
+	}
+	StartIPInt := binary.BigEndian.Uint32(StartIPRaw.To4())
+	EndIPRaw := net.ParseIP(EndIP).To4()
+	if len(EndIPRaw) == 0 || len(EndIPRaw) > net.IPv4len {
+		return nil, fmt.Errorf("range is Invalid")
+	}
+	EndIPInt := binary.BigEndian.Uint32(EndIPRaw.To4())
+	if StartIPInt > EndIPInt {
+		return nil, fmt.Errorf("range is Invalid")
+	}
+	rangeIP := binary.BigEndian.Uint32(RandIP4().To4())
+	temp := EndIPInt - StartIPInt
+	if temp <= 0 {
+		return nil, fmt.Errorf("range is Invalid")
```

Comment:
This basically means the user is not allowed to specify the start & end as being the same IPs. We shouldn't throw an error in this case, since the range is inclusive, not invalid. Yes, the range only includes 1 IP, but it's still a valid range, regardless of how dumb that sounds.

So in that special case, instead of throwing an error, we should simply return the input `startIP` as a `net.IP`. We performed this parse above but we discarded it because we said `.To4()` on it. We should save the intermediate variable between those and use it here.

Reply:
Also, this means that if this case is entered, the `rangeIP` above becomes useless. So we should only calculate `rangeIP` below this case, when we start manipulating it.